drivers/sensor/ltrf216a/ltrf216a.c

```diff
@@ -0,0 +1,220 @@
+				      uint32_t size)
+{
+	const struct ltrf216a_config *config = dev->config;
+
+	return i2c_burst_read_dt(&config->i2c, reg, val, size);
+}
+
+static int ltrf216a_reg_write(const struct device *dev, uint8_t reg, uint8_t val)
+{
+	const struct ltrf216a_config *config = dev->config;
+
+	return i2c_write_dt(&config->i2c, &val, 1);
+}
+
+static int ltrf216a_enable(const struct device *dev)
+{
+	return ltrf216a_reg_write(dev, LTRF216A_MAIN_CTRL, LTRF216A_ALS_ENABLE_MASK);
+}
+
+static int ltrf216a_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	struct ltrf216a_data *drv_data = dev->data;
+	uint8_t status;
+	uint8_t value[LTRF216A_NUMBER_DATA_REGISTERS];
+	int result;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL || chan == SENSOR_CHAN_LIGHT);
+
+	drv_data->sample[0] = 0U;
+	drv_data->sample[1] = 0U;
+	drv_data->sample[2] = 0U;
+
+	result = ltrf216a_enable(dev);
+	if (result != 0) {
+		LOG_ERR("ltfr216a: enable failed");
+		return result;
+	}
+
+	while ((result = ltrf216a_reg_read(dev, LTRF216A_MAIN_STATUS, &status)) == 0) {
+		if ((status & LTRF216A_ALS_DATA_STATUS) != 0) {
+			break;
+		}
+
+		k_sleep(K_MSEC(LTRF216A_ALS_READ_DATA_DELAY_MS));
+	}
+
+	if (ltrf216a_reg_read_multiple(dev, LTRF216A_ALS_CLEAR_DATA_0, value,
+				       LTRF216A_NUMBER_DATA_REGISTERS) != 0) {
+		LOG_ERR("ltfr216a: read multiple failed");
+		return -EIO;
+	}
+	drv_data->sample[0] = value[3];
+	drv_data->sample[1] = value[4];
+	drv_data->sample[2] = value[5];
+
+	return 0;
+}
+
+static int ltrf216a_channel_get(const struct device *dev, enum sensor_channel chan,
+				struct sensor_value *val)
+{
+	struct ltrf216a_data *drv_data = dev->data;
+
+	if (chan != SENSOR_CHAN_LIGHT) {
+		return -ENOTSUP;
+	}
+
+	int greendata = sys_get_le24(drv_data->sample);
+
+	uint64_t lux, divisor;
+	/*
+	 * 0.45 -> 45/100, multiplied by 100 for numeric division
+	 */
+	lux = greendata * 45 * 100 * LTRF216A_WIN_FAC;
+	/*
+	 * gain 3 (default), integration time 100ms=1, 1/100 from 45, 100 from numeric division
+	 */
+	divisor = 3 * 1 * 100 * 100;
+	uint64_t calc = lux / divisor;
+
+	val->val1 = calc;
+	val->val2 = 0;
+
+	return 0;
+}
+
+static const struct sensor_driver_api ltrf216a_driver_api = {
+	.sample_fetch = ltrf216a_sample_fetch,
+	.channel_get = ltrf216a_channel_get,
+};
+
+static int ltrf216a_chip_init(const struct device *dev)
+{
+	const struct ltrf216a_config *config = dev->config;
+	uint8_t value;
+
+	if (!i2c_is_ready_dt(&config->i2c)) {
+		LOG_ERR("Bus device is not ready");
+		return -ENODEV;
+	}
+
+	if (ltrf216a_reg_read(dev, LTRF216A_PART_ID, &value) != 0) {
+		return -EIO;
+	}
+
+	if (((value & LTRF216A_PART_ID_PARTNUMBERID_MASK) >> LTRF216A_PART_ID_PARTNUMBERID_SHIFT !=
+	     LTRF216A_PART_ID_PARTNUMBERID_VALUE) ||
+	    ((value & LTRF216A_PART_ID_REVISIONID_MASK) != LTRF216A_PART_ID_REVISIONID_VALUE)) {
+		LOG_ERR("Bad manufacturer id 0x%x", value);
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+int ltrf216a_init(const struct device *dev)
```

Comment:
You don't need this wrapper function, you can rename ltrf216a_chip_init and use that directly.

Reply:
applied in the latest force push.